saleor/plugins/avatax/plugin.py

```diff
@@ -373,29 +372,64 @@ def calculate_checkout_line_total(
 
         return base_total
 
-    def _calculate_order_line_unit(self, order_line):
+    def calculate_checkout_line_unit_price(
+        self,
+        checkout: "Checkout",
+        checkout_line: "CheckoutLine",
+        address: Optional["Address"],
+        discounts: Iterable["DiscountInfo"],
+        variant: "ProductVariant",
+        collections: List["Collection"],
+        channel: "Channel",
+        channel_listing: "ProductVariantChannelListing",
+        previous_value: TaxedMoney,
+    ):
+        if not checkout_line.variant.product.charge_taxes:
+            return previous_value
+        return self._calculate_unit_price(
+            checkout, checkout_line, variant, previous_value, discounts, is_order=False
+        )
+
+    def calculate_order_line_unit(
```

Comment:
Can we pass Variant and product to this function instead of getting them from the order line? 
Your solution can generate a lot of database queries.

Reply:
We can pass `CheckoutLineInfo` too.